### PR TITLE
Integreatly OCP4 role refactor

### PR DESCRIPTION
**Pull request by aidenkeating**
##### SUMMARY

Refactor Integreatly roles for OCP4 to have the:
- Flat variable naming, instead of dicts
- Snake case role naming

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
 
Roles:
- ocp4_workload_integreatly
- ocp4_workload_integreatly_minio

##### ADDITIONAL INFORMATION

- Run against an OCP 4.3 Workshop
- Ensure RHMI is installed successfully after role completes

### CODE REVIEW

**Comment by aidenkeating** [comment]
@wkulhanek Made changes based on your suggestions (flat var names, snake case role names). I wasn't sure whether to move the Integreatly roles themselves into `roles_ocp_workloads`?

File: ansible/roles/ocp4_workload_integreatly/tasks/workload.yml [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  k8s:
+    state: present
+    namespace: "{{ ocp4_workload_integreatly_namespace }}"
+    merge_type:
+    - strategic-merge
+    - merge
+    definition: "{{ item }}"
+  loop: "{{ _operator_manifest_crd_paths }}"
+
+- name: Create workshop RHMI resource
```
Comment (pmccarthy) [inline ansible/roles/ocp4_workload_integreatly/tasks/workload.yml:115]: @aidenkeating Is there a need to validate that Minio is available before trying to install an instance of RHMI? Or is it a case that by the time the integreatly operator gets around to configuring 3Scale, Minio would have had plenty of time to come up by then?

**Comment by newgoliath** [comment]
Yes, put them there.  That way they're sorted away from Ansible and others.

File: ansible/roles/ocp4_workload_integreatly/tasks/workload.yml (hunk 1 above)
Comment (newgoliath) [reply ansible/roles/ocp4_workload_integreatly/tasks/workload.yml:115]: The Minio workload should not exit cleanly until minio is deployed OK.

File: ansible/roles/ocp4_workload_integreatly/sample_vars/ocp4_workload_integreatly_vars.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+
```
Comment (newgoliath) [inline ansible/roles/ocp4_workload_integreatly/sample_vars/ocp4_workload_integreatly_vars.yml:2]: add comments.  This is where users will interface.

File: ansible/roles/ocp4_workload_integreatly/sample_vars/ocp4_workload_integreatly_vars.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ocp4_workload_integreatly_user_count: 50
+ocp4_workload_integreatly_user_password: "{{ lookup('password', '/tmp/intly-userpass length=16 chars=ascii_letters,digits') }}"
+ocp4_workload_integreatly_dedicated_admin_user_password: "{{ lookup('password', '/tmp/intly-dauserpass length=16 chars=ascii_letters,digits') }}"
+ocp4_workload_integreatly_admin_user_password: "{{ lookup('password', '/tmp/intly-adminpass length=16 chars=ascii_letters,digits') }}"
+
+# override ocp workloads to include prerequisite roles
+ocp_workloads:
+  - ocp4_workload_integreatly_minio
+  - ocp4-workload-authentication
```
Comment (newgoliath) [inline ansible/roles/ocp4_workload_integreatly/sample_vars/ocp4_workload_integreatly_vars.yml:11]: Use the new role location

File: ansible/roles/ocp4_workload_integreatly/tasks/remove_workload.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+# Implement your Workload removal tasks here
+# ------------------------------------------
+
```
Comment (newgoliath) [inline ansible/roles/ocp4_workload_integreatly/tasks/remove_workload.yml:4]: No AWS resources of any kind were created that need cleanup?
S3?  DNS?  VIP?

Comment (aidenkeating) [reply ansible/roles/ocp4_workload_integreatly/tasks/remove_workload.yml:4]: @newgoliath  It doesn't, no, the `ocp4_workload_integreatly` role only creates resources in the cluster

File: ansible/roles/ocp4_workload_integreatly/tasks/workload.yml (hunk 1 above)
Comment (aidenkeating) [reply ansible/roles/ocp4_workload_integreatly/tasks/workload.yml:115]: @pmccarthy Minio is deployed successfully by the time we reach this stage

File: ansible/roles/ocp4_workload_integreatly/defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+# 3scale
+ocp4_workload_integreatly_threescale_namespace: redhat-rhmi-3scale
+ocp4_workload_integreatly_seed_secret_name: system-seed
+ocp4_workload_integreatly_threescale_tenant_admin_user: admin
+ocp4_workload_integreatly_threescale_tenant_admin_email: admin@example.com
+ocp4_workload_integreatly_threescale_tenant_admin_password: Password1
+ocp4_workload_integreatly_threescale_tenant_detail_secret_name: admin-tenant
+ocp4_workload_integreatly_threescale_tenant_detail_secret_namespace: redhat-rhmi-3scale
+ocp4_workload_integreatly_threescale_tenant_id_format: evals%02d-tenant
+ocp4_workload_integreatly_threescale_users_name_format: evals%02d
```
Comment (austincunningham) [inline ansible/roles/ocp4_workload_integreatly/defaults/main.yml:47]: been using this var for fuse as well, maybe better to move it to the operator section

File: ansible/roles_ocp_workloads/ocp4_workload_integreatly/tasks/post_workload.yml [hunk 6]
```diff
@@ -57,19 +74,18 @@
...
+  shell: oc adm groups add-users dedicated-admins {{ ocp4_workload_integreatly_dedicated_admin_username }}
   when: group_output.rc != 1
 
 - name: Print Overview
   debug:
     msg: "{{ item }}"
   with_items:
-    - "user.info: Dedicated Admin User: {{ ocp4_workload_integreatly.dedicated_admin_username }} / {{ _dedicated_admin_password }} (This is the most privileged user account available to consultants and customers in RHMI production/poc environments.)"
-
+    - "user.info: Dedicated Admin User: {{ ocp4_workload_integreatly_dedicated_admin_username }} / {{ ocp4_workload_integreatly_dedicated_admin_user_password }} (This is the most privileged user account available to consultants and customers in RHMI production/poc environments.)"
```
Comment (pmccarthy) [inline ansible/roles_ocp_workloads/ocp4_workload_integreatly/tasks/post_workload.yml:88]: @aidenkeating We should probably include more details here. Similar to what we have for 1.x: https://github.com/redhat-cop/agnosticd/blob/development/ansible/roles/ocp-workload-integreatly/tasks/workload.yml#L55-L62

Can rename "Web App URL" to "Solution Explorer URL" instead.

**Comment by aidenkeating** [comment]
@wkulhanek @pmccarthy Updated with:
- Supported versions file
- Cleaned up output
- Avoid recreating tenants in 3scale

**Pull request merged**